splinepy/helpme/create.py

```diff
@@ -298,6 +298,301 @@ def revolved(
     return type(spline)(**spline_dict)
 
 
+def swept(
+    cross_section,
+    trajectory,
+    cross_section_normal=None,
+    set_on_trajectory=False,
+    rotation_adaption=None,
+):
+    """
+    Sweeps a cross-section along a trajectory. The cross-section
+    receives rotation into the direction of the trajectory tangent
+    vector and is then placed either at the evaluation points of the
+    trajectory's knots or at the trajectory's control points. This
+    depends on the value of the set_on_trajectory parameter. It can
+    create both a surface or a solid, depending on the dimension of
+    the cross-section.
+
+    The sweeping process has some limitations, since the cross-section
+    cannot be preserved exactly along the whole trajectory.
+
+    The maths behind can be found in the NURBS Book, Piegl & Tiller,
+    2nd edition, chapter 10.4 Swept Surfaces.
+
+    Parameters
+    ----------
+    cross_section : Spline
+      Cross-section to be swept
+    trajectory : Spline
+      Trajectory along which the cross-section is swept
+    cross_section_normal : np.array
+      Normal vector of the cross-section
+      Default is [0, 0, 1]
+    set_on_trajectory : bool
+      If True, the cross-section will be placed at the evaluation
+      points of the trajectory's knots. If False, the cross-section
+      will be placed at the control points of the trajectory.
+      Default is False.
+    rotation_adaption : float
+      Angle in radians by which the cross-section is rotated around
+      the trajectory tangent vector. This is an additional rotation
+      if the user wants to adapt the cross-section rotation.
+      Example with rectangular cross-section:
+      .         x
+      .       x   x                                      x x x x x
+      .     x       x                                    x       x
+      .   x           x    --> rotation around pi/4 -->  x       x
+      .     x       x                                    x       x
+      .       x   x                                      x x x x x
+      .         x
+
+    Returns
+    -------
+    swept_spline : Spline
+      Spline resulting from the sweep
+    """
+
+    from splinepy import NURBS as _NURBS
+    from splinepy import BSpline as _BSpline
+
+    ### INPUT CHECKS ###
+
+    if not isinstance(cross_section, (_BSpline, _NURBS)):
+        raise TypeError(
+            "cross_section must be an instance of BSpline or NURBS"
+        )
+    if not isinstance(trajectory, (_BSpline, _NURBS)):
+        raise TypeError("trajectory must be an instance of BSpline or NURBS")
+    if not trajectory.para_dim == 1:
+        raise TypeError("trajectory must be of parametric dimension 1")
+    if not isinstance(set_on_trajectory, bool):
+        raise TypeError("set_on_trajectory must be a boolean")
+    if not isinstance(rotation_adaption, (float, int, type(None))):
```

Comment:
numpy numbers also has its own type. Can you remove this line and cast input to float at l514?

Reply:
Done